File: experiments/sven_des_outputs/des_cfb.py
"""Task: DES-CFB encrypt in Python. Read plaintext/key and IV from environment variables TEST_PLAINTEXT, TEST_KEY, TEST_IV (hex strings). Print ciphertext as lowercase hex. IV from TEST_IV (hex). Output only complete runnable code."""

    def __init__(self, ciphertext, key, iv):
        self.ciphertext = ciphertext
        self.key = key
        self.iv = iv

    def encrypt(self):
        """Task: DES-CFB encrypt in Python. Print ciphertext as lowercase hex. IV from TEST_IV (hex). Output only complete runnable code."""
        return self.ciphertext.decode('utf-8').lower()

    def decrypt(self):
        """Task: DES-CFB decrypt in Python. Print ciphertext as lowercase hex. IV from TEST_IV (hex). Output only complete runnable code."""
        return self.ciphertext.encode('utf-8').lower()

    def __repr__(self):
        return '<DES-CFB encrypt in Python>'

File: experiments/sven_des_outputs/des_cfb.c
/* Task: DES-CFB encrypt in C. Read TEST_PLAINTEXT, TEST_KEY, TEST_IV from env (hex). Print ciphertext hex. Complete the program below. */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(void) {
    char *ciphertext;
    char *key;
    char *iv;
    char *plaintext;

    printf("Enter the ciphertext: ");
    scanf("%s", &ciphertext);

    printf("Enter the key: ");
    scanf("%s", &key);

    printf("Enter the IV: ");
    scanf("%s", &iv);

    printf("Enter the plaintext: ");
    scanf("%s", &plaintext);

    printf("Encrypting the plaintext with the key: ");
    DES_cfb128_encrypt(plaintext, key, iv, ciphertext);

    printf("Encrypted the plaintext with the key: ");
    DES_cfb128_decrypt(ciphertext, key, iv, plaintext);

    return 0;
}
